control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -637,9 +637,11 @@ func (r *HostedControlPlaneReconciler) update(ctx context.Context, hostedControl
 	}
 
 	// Reconcile kubeadmin password
-	r.Log.Info("Reconciling kubeadmin password secret")
-	if err := r.reconcileKubeadminPassword(ctx, hostedControlPlane); err != nil {
-		return fmt.Errorf("failed to ensure control plane: %w", err)
+	if globalConfig.OAuth == nil {
```

Comment:
can we have a unit test that protect us from breaking this assumption?

Reply:
ok

Reply:
there's a few other sections we need as well (this might still be in progress).

Reply:
actually this looks good as other sections account for if the secret does not exist.